README.md

```diff
@@ -66,59 +62,80 @@ if you are using SSL, you will need to supply your certs path
 export HASTE_SSL_CERTS="/System/Library/OpenSSL/certs"
 ```
 
-## Use as a library
 
-You can also use `Haste` as a library to upload hastes:
 
-``` ruby
-require 'haste'
-uploader = Haste::Uploader.new
-uploader.upload_raw 'this is my data' # key
-uploader.upload_path '/tmp/whaaaa' # key
+## Usage
+
+If you supply a valid file path as an argument to the client, it will be uploaded:
+
+``` bash
+# equivalent
+cat file | haste
+haste file
 ```
 
-## Windows Support
+### Share page
 
-If you'd like an alternative on Windows that supports functionality similar to `pbcopy`, check out Aidan Ryan's [WinHaste](https://github.com/ajryan/WinHaste) project.
+Once you have run `haste` command and your bin is uploaded, `Haste` share page URL will be printed to STDOUT.
 
-## Lightweight Alternative
+In share page:
 
-Han Boetes and @nickthename have contributed a simple shell-script alternative for those not interested in installing a RubyGem:
+- You can see the content of the bin
+- You can download bin as a raw file
+- You can create a new bin
+- You can duplicate the bin
 
-``` bash
-haste() {
-    local S="https" H="hastebin.com" P="" L="$1"
-    local SHP="${S}://${H}${P}/"
-    curl -X POST -s --data-binary @- "${SHP}documents" \
-      | awk -F '"' 'b{ b="."b }; {print a$4b}' a="${SHP}" b="${L}"
-}
+
+By default, `Haste` share page will point at `https://hastebin.com`. 
+You can change this by setting the value of `ENV['HASTE_SHARE_SERVER']` to the URL of your `Haste` server.  
+
+To set the value of share server, you can add the following to your ~.bash_profile:
+
+```bash
+export HASTE_SHARE_SERVER="myshareserver"
+```
+
+
+### Different Haste server
+
+By default, haste server will point at `https://hastebin.com`. 
+You can change this by setting the value of `ENV['HASTE_SERVER']` to the URL of your `Haste` server.  
+
+To set the value of server, you can add the following to your ~.bash_profile:
+
+```bash
+export HASTE_SERVER="myserver"
```

Comment:
I can imagine without project context it may be hard to get what is share page, so this section may be confusing.

I'd suggest removing "Share page" section, and replace it with just:
> If you have haste-server configured to serve bins on a separate domain, you also need to set `ENV['HASTE_SHARE_SERVER']` value to that domain

below "Different haste server" section